### ISSUE TITLE
Output of the clean_names function has changed

### ISSUE THREAD

**Issue by MislavSag** [sfirke/janitor#376]
## Bug reports

I updated my R to version 4.0. After that, the function `clean_names` changed the output. Before when I called the function on df that contains UTF-8 characters in column names, it worked as expected. For example, character č was changed to c, but now it converts č to e. More concretely, it converts stečajni to steecajni, but before it was stecajni. Is this change done with purpose or is something changes with new version of R?

**Comment by billdenney** [comment]
The change was intentional, and it is in janitor, not R.  But, it sounds like it is not working optimally for you, and I do think that it would be preferred to make the change to c instead of e in your example.

The new transliteration is done here using what is published as the preferred conversion to ASCII: https://github.com/sfirke/janitor/blob/ecc9f4e8413b1e396f20f08a08b43ef130211568/R/make_clean_names.R#L103

Perhaps the first transliteration should remove accent marks and then subsequent transliterations should work from there.

**Comment by MislavSag** [comment]
Thanks for the feedback. How can 'remove accent' in first transliteration ?

**Comment by billdenney** [comment]
That isn't currently possible; it would require a change to janitor.  Based on http://userguide.icu-project.org/transforms/general, I think that we would need to change the transliterators to include "NFD; [:Nonspacing Mark:] Remove; NFC", but I think that will change other aspects in a minor way that may be problematic.

@MislavSag, what locale are you working in?  (Can you please provide the output of `Sys.getlocale()`?)  I'm having trouble getting the č character to work in my R session as it becomes a regular "c" when I try `foo <- "č"`; I'm guessing that is a locale issue.

@MislavSag, can you please try the following code on your system as well and post the output of that, too?  Does it give the expected "c"?

```r
library(stringi)
foo <- "stečajni"
stri_trans_general(foo, "NFD; [:Nonspacing Mark:] Remove; NFC")
```

@gagolews, do you have any thoughts on the best way to fix this issue so that output is more like the expected output (where the accented c is converted to a regular c)?  (Is that suggestion above the right one?  Should We use NFKD and NFKC instead?  I'm a bit out of my depth to have confidence that the answer here is the right one.)

@sfirke, Fixing this will be another breaking change, but in my opinion, this is more of a bug fix for the new way of working rather than a new-major-version breaking change.

**Comment by MislavSag** [comment]
@billdenney 
My local:
`[1] "LC_COLLATE=Croatian_Croatia.1250;LC_CTYPE=Croatian_Croatia.1250;LC_MONETARY=Croatian_Croatia.1250;LC_NUMERIC=C;LC_TIME=Croatian_Croatia.1250"`

Your code returns 'e' again.

**Comment by billdenney** [comment]
Hmm, I'm surprised that it still generates an e.  Does the č sound like e when spoken in Croatian?  I ask because I'm surprised by the transliteration to e, and I'm trying to understand why it happens.

(Unrelated: I really want to visit Croatia some day.)

**Comment by MislavSag** [comment]
No, č doesn't sound like e. I can post the question on stringi repo.

(I am subjective, but I like the Croatian cost the most :) You would enjoy it.)

**Comment by gagolews** [comment]
`stri_trans_general()` is locale-agnostic. Otherwise, it would have the `locale` argument.

I get:

```
> stri_trans_general(foo, "NFD; [:Nonspacing Mark:] Remove; NFC")
[1] "stecajni"
```


Could  you try:

*  `stri_trans_general(foo, "NFKC; [:Nonspacing Mark:] Remove; NFC")`
* `stri_trans_general(foo, "Latin-ASCII")`

Also, can you report the retval of your `stri_info(TRUE)`?

**Comment by billdenney** [comment]
When I tried this on my system, I am getting "c" and not "e".  Unicode can be difficult that characters that are visually the same may have different code points (integer values).  With the conversions that janitor uses (the third example below), I appear to get a "c" and not an "e".

@MislavSag, in addition to what @gagolews is suggesting, can you please also run `utf8ToInt()` on the character that is giving you trouble to ensure that we are all working with the same character?

```r
library(stringi)
foo <- intToUtf8(269L)
foo
# [1] "č"
stri_trans_general(foo, "NFKC; [:Nonspacing Mark:] Remove; NFC")
# [1] "č"
stri_trans_general(foo, "Latin-ASCII")
# [1] "c"
stri_trans_general(foo, "Any-Latin;Greek-Latin;Latin-ASCII")
# [1] "c"
```

**Comment by sfirke** [comment]
@billdenney do you think there is follow-up to be done here?

**Comment by sfirke** [comment]
Sounds good to me 👍
